### PR TITLE
Enforce the baseline checkImplicitDependencies check

### PR DESCRIPTION
**Pull request by carterkozak**
**Goals (and why)**:

This should prevent atlas from breaking when it's consumed by products that use newer versions of shared dependencies.

**Implementation Description (bullets)**:

Builds must be updated

**Testing (What was existing testing like?  What have you done to improve it?)**:

This adds a test.

**Concerns (what feedback would you like?)**:

**Where should we start reviewing?**:

**Priority (whenever / two weeks / yesterday)**:

yesterday: This would have prevented several failures that made it through testing.

### CODE REVIEW

**Comment by pkoenig10** [comment]
Probably also want to enable `checkUnusedDependencies`?

**Review by Jolyon-S — APPROVED** [review]
Looks sensible - although we should probably enable `checkUnusedDependencies` as @pkoenig10 notes to ensure that we aren't depending on more than we need.

File: commons-executors-api/build.gradle [hunk 1]
```diff
@@ -5,3 +5,6 @@ libsDirName = file('build/artifacts')
 // Do not add dependencies to this project. This project is used by
 // api projects outside of atlasdb that try very hard to minimize
 // their dependency footprint.
+dependencies {
+    implementation 'com.boundary:high-scale-lib'
+}
```
Comment (jeremyk-91) [inline commons-executors-api/build.gradle:10]: given the comment above, are we sure this is ok?

**Review by jeremyk-91 — COMMENTED** [review]
seems good, one question though

File: commons-executors-api/build.gradle (hunk 1 above)
Comment (jeremyk-91) [reply commons-executors-api/build.gradle:10]: I guess it's fine as this was implicitly added anyway, but is there a way to avoid this?

**Comment by pkoenig10** [comment]
Are we still planning to merge this PR?

**Pull request merged**